src/main/java/com/ably/kafka/connect/ChannelSinkTask.java

```diff
@@ -32,28 +34,24 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import io.ably.lib.realtime.AblyRealtime;
-import io.ably.lib.realtime.Channel;
-import io.ably.lib.types.AblyException;
-import io.ably.lib.types.Message;
-import io.ably.lib.types.MessageExtras;
-import io.ably.lib.util.JsonUtils;
-import io.ably.lib.util.JsonUtils.JsonUtilsObject;
-import io.ably.lib.util.Log.LogHandler;
+import java.util.Base64;
+import java.util.Collection;
+import java.util.Map;
 
 public class ChannelSinkTask extends SinkTask {
     private static final Logger logger = LoggerFactory.getLogger(ChannelSinkTask.class);
     private static final String[] severities = new String[]{"", "", "VERBOSE", "DEBUG", "INFO", "WARN", "ERROR", "ASSERT"};
 
-    ChannelSinkConnectorConfig config;
-    AblyRealtime ably;
-    Channel channel;
+    private AblyRealtime ably;
+
+    private ChannelSinkMapping channelSinkMapping;
 
     @Override
     public void start(Map<String, String> settings) {
         logger.info("Starting Ably channel Sink task");
 
-        config = new ChannelSinkConnectorConfig(settings);
+        final ChannelSinkConnectorConfig config = new ChannelSinkConnectorConfig(settings);
+        channelSinkMapping = new DefaultChannelSinkMapping(config);
```

Comment:
Yes of course, this one is only for the case that a custom mapping class isn't provided. I have made it so that it can be provided as part of configuration
https://github.com/ably/kafka-connect-ably/blob/5d9c04d09ba0c52fa9a5086b7cfda75bf634a147/src/main/java/com/ably/kafka/connect/ChannelSinkConnectorConfig.java#L206

I will also consider the factory, depending on what the best way users can inject their own custom config classes 👍